core/recall.py

```
@@ -0,0 +1,342 @@
+    ) or is_transient_recall_error(error):
+        return TransientRecallError(str(error))
+    return error
+
+
+RecallFn = Callable[[SendReceipt], Awaitable[None]]
+
+RECALL_RETRY_DELAY_SECONDS = 1.0
+
+
+class RecallRegistry:
+    """持有自动撤回任务强引用，并负责统一取消与异常隔离。"""
```

Ruff (0.16.2)

[warning] Docstring contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF002)